### PR TITLE
#6449: DatePicker: Metadata add styleClass

### LINKED ISSUE #6449: DatePicker: Metadata: add styleClass

### PR DESCRIPTION
**Pull request by jepsar**
Fix #6449

### CODE REVIEW

File: docs/9_0/components/datepicker.md [hunk 1]
```diff
@@ -293,11 +293,12 @@ private DateMetadataModel lazyModel;
 @PostConstruct
 public void init() {
     DefaultDateMetadata metadataDisabled = DefaultDateMetadata.builder().disabled(true).build();
+    DefaultDateMetadata metaDataDeadline = DefaultDateMetadata.builder().styleClass("deadline").build();
```
Comment (tandraschko) [inline docs/9_0/components/datepicker.md]: Metadata

File: src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js [hunk 2]
```diff
@@ -302,12 +302,19 @@ PrimeFaces.widget.DatePicker = PrimeFaces.widget.BaseWidget.extend({
...
                         var disabledDates = [];
+                        var dateStyleClasses = {};
                         for (date in dateMetadata) {
+                            var parsedDate = pdp.parseOptionValue(date);
                             if (dateMetadata[date].disabled) {
-                                disabledDates.push(date);
+                                disabledDates.push(parsedDate);
+                            }
+                            if (dateMetadata[date].styleClass) {
+                              dateStyleClasses[pdp.toISO(parsedDate)] = dateMetadata[date].styleClass;
```
Comment (melloware) [inline src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js]: Check your spacing here it looks like this line needs 2 more space indent.

**Review by melloware — CHANGES_REQUESTED** [review]

File: src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js (hunk 2 above)
Comment (jepsar) [reply src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js]: I already fixed my setting this weekend. Will fix this one.

File: src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js [hunk 3]
```diff
@@ -1602,6 +1610,10 @@
             return _classes;
         },
 
+        toISO: function (date) {
```
Comment (melloware) [inline src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js]: JSDoc this method what it returns and why you are only returning first 10 chars.

**Review by melloware — CHANGES_REQUESTED** [review]

File: src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js (hunk 3 above)
Comment (melloware) [reply src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js]: Maybe call it `toISOString(date)`

Comment (jepsar) [reply src/main/resources/META-INF/resources/primefaces/datepicker/0-datepicker.js]: > Maybe call it `toISOString(date)`

I'll rename it to `toISODateString` and add documentation. Basically we just care about the date part here and not about the time, so I'm taking only the date part.

**Review by melloware — APPROVED** [review]

File: src/main/java/org/primefaces/model/datepicker/DefaultDateMetadata.java [hunk 4]
```diff
@@ -62,12 +75,15 @@ public boolean equals(Object obj) {
             return false;
         }
         final DefaultDateMetadata other = (DefaultDateMetadata) obj;
-        return this.disabled == other.disabled;
+        if (this.disabled != other.disabled) {
+            return false;
+        }
+        return Objects.equals(this.styleClass, other.styleClass);
     }
```
Comment (melloware) [inline src/main/java/org/primefaces/model/datepicker/DefaultDateMetadata.java:76]: Why not just make your equals and hashcode methods...
```java
@Override
    public int hashCode() {
        return Objects.hash(disabled, styleClass);
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (!(obj instanceof DefaultDateMetadata)) {
            return false;
        }
        DefaultDateMetadata other = (DefaultDateMetadata) obj;
        return disabled == other.disabled && Objects.equals(styleClass, other.styleClass);
    }
```

**Review by melloware — CHANGES_REQUESTED** [review]

File: src/main/java/org/primefaces/model/datepicker/DefaultDateMetadata.java (hunk 4 above)
Comment (jepsar) [reply src/main/java/org/primefaces/model/datepicker/DefaultDateMetadata.java:76]: I did not bother writing them. It's how they were generated by NetBeans.

Comment (melloware) [reply src/main/java/org/primefaces/model/datepicker/DefaultDateMetadata.java:76]: Ahh mine above were generated by Eclipse.  Interesting!

Comment (jepsar) [reply src/main/java/org/primefaces/model/datepicker/DefaultDateMetadata.java:76]: .. doesn't matter, I'll change them

Comment (melloware) [reply src/main/java/org/primefaces/model/datepicker/DefaultDateMetadata.java:76]: OK cool yeah its juts that `Objects.hash()` already does all the magic inside of it so its strange Netbeans generates like that.

File: docs/9_0/components/datepicker.md (hunk 1 above)
Comment (jepsar) [reply docs/9_0/components/datepicker.md]: Ah, that slipped in here by copy pasting it from the showcase.. let me fix it there as well

**Review by melloware — APPROVED** [review]

**Comment by tandraschko** [comment]
Am i blind or isnt loadInitialLazyMetadata called?

**Comment by jepsar** [comment]
> Am i blind or isnt loadInitialLazyMetadata called?

It is called:

https://github.com/primefaces/primefaces/pull/6450/files?diff=unified&w=1#diff-6d9ed21fdb2d6506a99c1130235119aa694d1bb61118c0a0e715a93bcc1bd411R209

**Pull request merged**